cpp/dolfinx/io/ADIOS2Writers.cpp

```diff
@@ -702,16 +702,18 @@ void ADIOS2Writer::close()
 //-----------------------------------------------------------------------------
 FidesWriter::FidesWriter(MPI_Comm comm, const std::filesystem::path& filename,
                          std::shared_ptr<const mesh::Mesh> mesh)
-    : ADIOS2Writer(comm, filename, "Fides mesh writer", mesh)
+    : ADIOS2Writer(comm, filename, "Fides mesh writer", mesh),
+      _reuse_mesh(false)
 {
   assert(_io);
   assert(mesh);
   fides_initialize_mesh_attributes(*_io, *mesh);
 }
 //-----------------------------------------------------------------------------
 FidesWriter::FidesWriter(MPI_Comm comm, const std::filesystem::path& filename,
-                         const ADIOS2Writer::U& u)
-    : ADIOS2Writer(comm, filename, "Fides function writer", u)
+                         const ADIOS2Writer::U& u, const bool reuse_mesh)
```

Comment:
I never find `bool` argument very expressive. Maybe an `enum` would be better?

Reply:
It's a binary choice so a `bool` looks semantically appropriate. I can of course put an `enum` if you like it better, something like `mesh_reuse_policy`.

Reply:
I introduced an `enum` with the same design as in `XDMFFile`.